ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LedgerPeers/Type.hs

```diff
@@ -51,7 +52,7 @@ isLedgerPeersEnabled UseLedgerPeers {}  = True
 --
 newtype PoolStake = PoolStake { unPoolStake :: Rational }
   deriving (Eq, Ord, Show)
-  deriving newtype (Fractional, Num, NFData)
```

Comment:
I think it would be better to write the instance by hand and document it on the CDDL spec.

Reply:
One wrinkle here is that is you look at [Slot](https://github.com/IntersectMBO/cardano-base/blob/master/cardano-slotting/src/Cardano/Slotting/Slot.hs#L40) in cardano-slotting, the SlotNo type, which is a component of the snapshot, has a newtype automatically derived instance for Serialise, which is leveraged by the ToCBOR instance. Same for WithOrigin. And so I might as well bite the bullet and keep my auto derived instances as well I think.

Reply:
So I've issued a PR to cardano-base so that hand written instances can be made for the WithOrigin type for recording the slot number of the snapshot.

Reply:
So I've hand written all instances necessary to serialise the snapshot.